### PR TITLE
Clamav mirror update

### PR DESCRIPTION
**Pull request by teng1**
Hi, 
There has been some recent updates and releases by the ClamAV team relating to how private mirrors can be hosted, concurrency of reloads, move to cloudlflare CDN and stricter rate limiting, so there are some changes here relating to those.

Description of changes made:
- Update to ClamAV 103.x now supports concurrent reload of signature definitions
- clamav-notify has been renamed to clamav-mirror and has internal scheduler / self test and signature roll out 
- clamav-mirror now uses https://github.com/Cisco-Talos/cvdupdate rather than freshclam
- patching of deployment via k8s cronjob /staged reloads removed 
- Service account and custom role bindings removed
- Prometheus metrics relating to scheduler published on mirror 
- Additional response handling added to http interface in v1alpha
- HTTP messages in v1alpha correlate to response messages
- horizontal pod autoscaler api updated.
- some additional configuration options added to values file (threads / timeouts)
- 24 Hour update bug fix
- misc version bumps
- 429 error bug fix (rate limiting from upstream Cisco-Talos CDN )  
- clamav-mirror is now a statefulset (to support HA mirrors)

### CODE REVIEW

File: clamav-mirror/test.sh [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+##############################################
+# Globals:
+#   FRESHCLAM_CONFIG
+# Arguments:
+#   command
+# Outputs:
+#   Writes log to stdout
+##############################################
+function testfreshclam() {
+  local command
```
Comment (nefischer) [inline clamav-mirror/test.sh]: is that needed? not sure what it does...

File: clamav-mirror/test.sh [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+##############################################
+# Basic test of eicar and clean file 
+# Outputs:
+#   Writes log to stdout
+##############################################
+function scantest() {
+  log "INFO: Testing for eicar"
+  if ! clamscan files/eicar.txt
```
Comment (nefischer) [inline clamav-mirror/test.sh:68]: A bit strange that there is a `!` in the condition, considering that there is an 'if' and an 'else' block.
I had to read it twice to make sure I followed the logic.
Just a comment. No need to change if you feel strongly about it...

File: clamav-mirror/test.sh (hunk 1 above)
Comment (teng1) [reply clamav-mirror/test.sh]: thanks, this is not used for anything

File: clamav-mirror/test.sh (hunk 2 above)
Comment (teng1) [reply clamav-mirror/test.sh:68]: Yeah the command will exit non-zero if a virus is found so is expected to fail in order to "pass", the `!` for error handling is  used quite a bit in a styleguide was trying to follow. I changed the log message below it though!

File: clamav-mirror/test.sh [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    then 
+      log "PASS"
+    else 
+      err "ERROR: Unexpected status, expected 1 got $?"
+      exit
+  fi
+  if ! clamscan files/safe.txt
+    then 
+      err "ERROR Unexpected status, expected 1 got $?"
```
Comment (nefischer) [inline clamav-mirror/test.sh]: expected 0?

Comment (nefischer) [reply clamav-mirror/test.sh]: For some reason, the update didn't appear in the PR UI. Been changed as discussed. Resolving

**Comment by nefischer** [comment]
LGTM

**Pull request merged**